### ISSUE TITLE
List deserialize with Json serializer throws

### ISSUE THREAD

**Issue by Xelloss83** [esskar/Serialize.Linq#96]
Using this class
```
public class Test
{
    public string Code { get; set; }
}
```

Following works fine:
```
public void SerializeDeserializeListAsBinary()
{
    var list = new List<string> { "one", "two" };
    Expression<Func<Test, bool>> expression = test => list.Contains(test.Code);

    var serializer = new ExpressionSerializer(new BinarySerializer())
    {
        AutoAddKnownTypesAsListTypes = true
    };
    var value = serializer.SerializeBinary(expression);

    var actualExpression = (Expression<Func<Test, bool>>)serializer.DeserializeBinary(value);
    var func = actualExpression.Compile();

    Assert.IsTrue(func(new Test { Code = "one" }), "one failed.");
    Assert.IsTrue(func(new Test { Code = "two" }), "two failed.");
    Assert.IsFalse(func(new Test { Code = "three" }), "three failed.");
}
```

But as soon as I change serializer to Json, I'm getting a **MissingMethodException** saying "_Constructor on type 'System.Collections.Generic.List`1[[System.String, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]]' not found._"

Here is the code:
```
public void SerializeDeserializeListAsJson()
{
    var list = new List<string> { "one", "two" };
    Expression<Func<Test, bool>> expression = test => list.Contains(test.Code);

    var serializer = new ExpressionSerializer(new JsonSerializer())
    {
        AutoAddKnownTypesAsListTypes = true
    };
    var value = serializer.SerializeText(expression);

    var actualExpression = (Expression<Func<Test, bool>>)serializer.DeserializeText(value);
    var func = actualExpression.Compile();

    Assert.IsTrue(func(new Test { Code = "one" }), "one failed.");
    Assert.IsTrue(func(new Test { Code = "two" }), "two failed.");
    Assert.IsFalse(func(new Test { Code = "three" }), "three failed.");
}
```

**Comment by esskar** [comment]
fixed it. need to push to nuget again

**Comment by esskar** [comment]
published, version 1.6.0 should be available soon via nuget
